src/event-manager.js

```diff
@@ -87,7 +87,10 @@ function handleDelegatedEvent(event) {
       }
     }
 
-    target = target.parentNode;
+    const parent = target.parentNode;
+    const parentIsShadowRoot = parent && parent instanceof ShadowRoot;
```

Comment:
@michaelw85 Sorry for this, but I think your previous resolution is better. on browsers without shadow DOM, this will throw as ShadowRoot is undefined. Unless we do a small check on global.

Reply:
I like the `instanceof` though. Could do this for safety:

```ts
const parentIsShadowRoot = typeof ShadowRoot !== 'undefined' && parent instanceof ShadowRoot;
```

No need for the truthy check, that's implied by `instanceof`

Reply:
With refactoring the original behavior will not hit the instanceof check anymore.